src/sidebar/components/toast-messages.js

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+import useStore from '../store/use-store';
+
+import SvgIcon from './svg-icon';
+
+function ToastMessage({ message }) {
+  const removeMessage = useStore(store => store.removeToastMessage);
+  // Capitalize the message type for prepending
+  const prefix = message.type.charAt(0).toUpperCase() + message.type.slice(1);
+
+  /**
+   * a11y linting is disabled here: There is a click-to-remove handler on a
+   * non-interactive element. This allows sighted users to get the toast message
+   * out of their way if it interferes with interacting with the underlying
+   * components. This shouldn't pose the same irritation to users with screen-
+   * readers as the rendered toast messages shouldn't impede interacting with
+   * the underlying document.
+   */
+  return (
+    /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */
+    <li
+      className="toast-message-container"
+      onClick={() => removeMessage(message.id)}
```

Comment:
I think we may need to keep tract of a local state variable such as `const [dismissMessage, setDismissMessage] = useState(false);`

My hack foo here was

```onClick={() => {
        setDismissMessage(true);
        setTimeout(()=>{
          removeMessage(message.id);
        }, 200); // enough delay to allow the css dismiss animation to finish
```

It's stupid simple but it delays setting the redux store by just enough time to allow the css close animation to finish, then it removes the component from the DOM via removeMessage process. Otherwise we don't see the closing animation.

Reply:
I  have a mild attachment to the lightness of the current implementation, in which we're not tracking things or attaching extra classnames to elements or having to track different shades of state...let me give this some thought.

Reply:
I could clarify, we dont see the closing animation when clicked -- which felt a bit unfinished to me so that is why I proposed keep state var.

The other thing that is a little rough for me is the dependency on the timeout from the service (5s) and the css animation delay (4.5s). They absolutely depend on each-other but there is nothing to indicate that. I sorta feel the proxy component to run fadin/out animation on a subcomponent feels right here, but I understand your reservations about that.

Maybe that could be an enhancement after this is landed?